clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -275,6 +275,13 @@ Response HandleFunction(Sema &SemaRef, const FunctionDecl *Function,
                                      TemplateArgs->asArray(),
                                      /*Final=*/false);
 
+    if (RelativeToPrimary &&
+        (Function->getTemplateSpecializationKind() ==
+             TSK_ExplicitSpecialization ||
```

Comment:
Is this condition `Function->getTemplateSpecializationKind() == TSK_ExplicitSpecialization` necessary? I removed it and all testcases passed. Could you please add a testcase to cover it which would fail without this condition?

Reply:
This testcase runs OK before [be79079](https://github.com/llvm/llvm-project/pull/88963/commits/be79079507ffbd9b29683498f405dc2c32dd8ba7) applied and it may not be related to the condition `Function->getTemplateSpecializationKind() == TSK_ExplicitSpecialization`.

Reply:
@jcsxky https://godbolt.org/z/7rTxcG7EE

Reply:
@sdkrystian but gcc fails on the testcase you provided.